### PR TITLE
Adding gaze divergence module

### PR DESCRIPTION
**Pull request by dunity**
### Purpose of this PR
Add functionality to any implementer, allowing them to detect when the HMD is focused upon a given transform, within a defined (degree-based)range.

The "foot-indicator-smooth-opacity" branch utilizes this module, in order to fade the PlayerFloor UI contents in/out, according to when they're being focused upon.

### Testing status
Tested in various scenes & conditions.  I haven't encountered issues with the addition of this Module.

### Technical risk
Low.  Outside of the addition of the module registration in EditorVR.cs, there's not much room for this code to cause conflicts.

### Comments to reviewers
The module's functionality can be tested with the "foot-indicator-smooth-opacity" branch (still in progress).

### CODE REVIEW

**Comment by StephenHodgson** [comment]
Please make sure this works with rect transforms too, thanks 😀

File: Scripts/Modules/GazeDivergenceModule/GazeDivergenceModule.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        {
+            m_GazeSourceTransform = CameraUtils.GetMainCamera().transform;
+        }
+
+        void Update()
+        {
+            var gazeRotationDifference = Quaternion.Angle(m_GazeSourceTransform.rotation, m_PreviousGazeRotation);
+            gazeRotationDifference *= gazeRotationDifference; // Square the difference for intended response curve/shape
+            m_GazeVelocity = m_GazeVelocity + gazeRotationDifference * Time.unscaledDeltaTime;
+            m_GazeVelocity = Mathf.Clamp01(m_GazeVelocity -= Time.unscaledDeltaTime);
```
Comment (AndrewTHEManeri) [inline Scripts/Modules/GazeDivergenceModule/GazeDivergenceModule.cs]: should this just be -?

Comment (dunity) [reply Scripts/Modules/GazeDivergenceModule/GazeDivergenceModule.cs]: Do you mean a null prop op check?

**Comment by dunity** [comment]
- Addressed @AndrewTHEManeri feedback
- Added summary to gazeStable property & class
- Cached the m_GazeSourceTransform's rotation in update

**Pull request merged**